cupy/manipulation/join.py

```diff
@@ -39,13 +39,18 @@ def concatenate(tup, axis=0):
         tup (sequence of arrays): Arrays to be joined. All of these should have
             same dimensionalities except the specified axis.
         axis (int): The axis to join arrays along.
+            If axis is None, arrays are flattened before use.
+            Default is 0.
 
     Returns:
         cupy.ndarray: Joined array.
 
     .. seealso:: :func:`numpy.concatenate`
 
     """
+    if axis is None:
+        tup = tuple(m.flatten() for m in tup)
```

Comment:
Could you fix as follows for the performance?
```suggestion
        tup = [m.ravel() for m in tup]
```